lib/get-editor-config.js

```diff
@@ -0,0 +1,230 @@
+          // in editorconfig) & should just be returned as regular strings.
+          value2 = String(value);
+        }
+        props[key2] = value2;
+      }
+    }
+    return props;
+  }
+  readConfigFilesSync(filepaths) {
+    const files = [];
+
+    filepaths.forEach(filepath => {
+      if (!fs.existsSync(filepath)) {
+        return;
+      }
+      let file;
+      try {
+        file = fs.readFileSync(filepath, 'utf8');
+      } catch (e) {
+        file = '';
+      }
+      files.push({
+        name: filepath,
+        contents: file,
+      });
+    });
+    return files;
+  }
+  processMatches(matches) {
+    // Set indent_size to 'tab' if indent_size is unspecified and
+    // indent_style is set to 'tab'.
+    if (
+      'indent_style' in matches &&
+      matches.indent_style === 'tab' &&
+      !('indent_size' in matches)
+    ) {
+      matches.indent_size = 'tab';
+    }
+
+    // Set tab_width to indent_size if indent_size is specified and
+    // tab_width is unspecified
+    if ('indent_size' in matches && !('tab_width' in matches) && matches.indent_size !== 'tab') {
+      matches.tab_width = matches.indent_size;
+    }
+
+    // Set indent_size to tab_width if indent_size is 'tab'
+    if ('indent_size' in matches && 'tab_width' in matches && matches.indent_size === 'tab') {
+      matches.indent_size = matches.tab_width;
+    }
+
+    return matches;
+  }
+  parseFromConfigs(configs, filepath) {
+    return this.processMatches(
+      configs.reverse().reduce((matches, file) => {
+        const pathPrefix = path.dirname(file.name);
+        file.contents.forEach(section => {
+          const [glob, options] = section;
+          if (!glob) {
+            return;
+          }
+          const fullGlob = this.buildFullGlob(pathPrefix, glob);
+          if (!this.fnmatch(filepath, fullGlob)) {
+            return;
+          }
+          matches = this.extendProps(matches, options);
+        });
+        return matches;
+      }, {})
+    );
+  }
+  resetEditorConfigCache() {
+    this.EDITOR_CONFIG_CACHE = null;
+  }
+  resetConfigResolutionsCache() {
+    this.CONFIG_RESOLUTIONS_CACHE.clear();
+  }
+  getConfigFileNames(filepath, options) {
+    const paths = [];
+    do {
+      filepath = path.dirname(filepath);
+      paths.push(path.join(filepath, options.config));
+    } while (filepath !== options.root);
+    return paths;
+  }
+  getConfigsForFiles(files) {
+    const configs = [];
+    for (const i in files) {
+      if (Object.prototype.hasOwnProperty.call(files, i)) {
+        const file = files[i];
+        const contents = this.parseString(file.contents);
+        configs.push({
+          name: file.name,
+          contents,
+        });
+        if ((contents[0][1].root || '').toLowerCase() === 'true') {
+          break;
+        }
+      }
+    }
+    return configs;
+  }
+  resolveEditorConfigFiles(_filepath = path.join(process.cwd(), 'dummy'), _options = {}) {
```

Comment:
be default it trying to resolve directory from file path, `provcess.cwd()` - directory, and resolution function return  `provcess.cwd()` parent folder (we don't need it), we add `dummy` to get correct output if filepath = directory

Reply:
provcess.cwd() = /root/path

dirname(/root/path) => root

but, we need to find .editorconfig under `/root/path`

Reply:
OK, can you change it to something that more closely resembles what we would actually do? Using "dummy" is confusing (since it happens to be the name of ember-cli addon's testing app).